Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -1189,3 +1213,26 @@ extension Sources {
         return swiftSources.count != relativePaths.count
     }
 }
+
+func computeMinimumDeploymentTarget(of binaryPath: AbsolutePath) throws -> PlatformVersion? {
+    let runResult = try Process.popen(arguments: ["xcrun", "vtool", "-show-build", binaryPath.pathString])
+    guard let versionString = try runResult.utf8Output().components(separatedBy: "\n").first(where: { $0.contains("minos") })?.components(separatedBy: " ").last else { return nil }
+    return PlatformVersion(versionString)
+}
+
+private func computeXCTestMinimumDeploymentTarget() -> PlatformVersion {
```

Comment:
Actually, I just realized that we shouldn't compute this here because PackageBuilder can run completely in-memory on a fake file system. This value should be input to PackageBuilder from somewhere else (probably via UserToolchain).

Reply:
Also, nothing about the XCTest logic is macOS-specific, by the way,.

If we need to care about Mac Catalyst/iOS/tvOS/watchOS in this context, then this function should take the platform/SDK as an input.

Reply:
I didn't realise that XCTest has a higher minimum than what we support building for on the other platforms as well, that's why this was done only for macOS. The question is what do we do for watchOS...

Reply:
I think the answer is actually to change this method to return the oldest supported version if we can't determine one via the XCTest.framework. This way we don't end up with a constant that needs updating.